lib/Graph/Graph.cpp

```diff
@@ -1131,6 +1131,36 @@ RescaleQuantizedNode *Function::createRescaleQuantized(llvm::StringRef name,
       new RescaleQuantizedNode(name, getParent()->uniqueType(*outTy), input));
 }
 
+Node *Function::createWeightedSum(llvm::StringRef name,
+                                  llvm::ArrayRef<NodeValue> data,
+                                  llvm::ArrayRef<float> weights) {
+  assert(data.size() == weights.size() &&
+         "Must have same number of data and weights.");
+  assert(data.size() > 0 && "No inputs provided.");
+
+  auto *outTy = data[0].getType();
```

Comment:
do we need to assert all data[i] have the same type?

Reply:
I expect that to hit an assert elsewhere (e.g. in `createMul()`) if it wasn't the case, but it can't hurt to add here too!